### PR TITLE
Tap changer extrema strategy behavior

### PR DESCRIPTION
**Pull request by Jerry-Jinfeng-Guo**
Test cases to verify the `min` and `max` tap changing strategy would actually return a tap that results in a minimum or maximum voltage even in the case of the whole voltage band range falls outside achievable voltage ranges. The same hold for the `any` strategy. This edge case is found to be a bug for `fast_any`.

### CODE REVIEW

File: tests/data/power_flow/automatic-tap-regulator/achievable-voltage-above-band-any/output.json
Comment (mgovers) [inline tests/data/power_flow/automatic-tap-regulator/achievable-voltage-above-band-any/output.json:1]: i think this file was erroneously checked in. should be `sym_output.json`

Comment (figueroa1395) [inline tests/data/power_flow/automatic-tap-regulator/achievable-voltage-above-band-any/output.json:1]: I think this file got here by mistake and should be deleted in favor of the already present `sym_input.json`. I believe this is also what CI is complaining about.

File: tests/data/power_flow/automatic-tap-regulator/achievable-voltage-above-band-fast-any/params.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  "calculation_method": "newton_raphson",
+  "tap_changing_strategy": "fast_any_tap",
+  "rtol": 1e-05,
+  "atol": {
+    "default": 1e-05,
+    "u": 100.0
+  },
+   "xfail": {
+    "raises": "AssertionError",
+    "reason": "Bug in fast any tap changing strategy ins this edge case"
```
Comment (mgovers) [inline tests/data/power_flow/automatic-tap-regulator/achievable-voltage-above-band-fast-any/params.json]:
```suggestion
    "reason": "Bug in fast any tap changing strategy in this edge case: https://github.com/PowerGridModel/power-grid-model/issues/1215"
```

File: tests/data/power_flow/automatic-tap-regulator/achievable-voltage-below-band-fast-any/params.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "calculation_method": "newton_raphson",
+  "tap_changing_strategy": "fast_any_tap",
+  "rtol": 1e-05,
+  "atol": {
+    "default": 1e-05,
+    "u": 100.0
+  },
+  "skip": true
```
Comment (mgovers) [inline tests/data/power_flow/automatic-tap-regulator/achievable-voltage-below-band-fast-any/params.json]: please use `xfail`

```suggestion
  "xfail": {
    "raises": "AssertionError",
    "reason": "Bug in fast any tap changing strategy in this edge case: https://github.com/PowerGridModel/power-grid-model/issues/1215"
  }
```

File: tests/data/power_flow/automatic-tap-regulator/achievable-voltage-above-band-any/params.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+  "calculation_method": "newton_raphson",
+  "tap_changing_strategy": "any_valid_tap",
```
Comment (figueroa1395) [inline tests/data/power_flow/automatic-tap-regulator/achievable-voltage-above-band-any/params.json:3]: Side note, we should probably add support for arrays of strings for the tap changing strategy in the validation tests, so that these 4 become just 2. Out of scope though, just worth mentioning.

File: tests/data/power_flow/automatic-tap-regulator/achievable-voltage-above-band-fast-any/params.json (hunk 1 above)
Comment (figueroa1395) [inline tests/data/power_flow/automatic-tap-regulator/achievable-voltage-above-band-fast-any/params.json]: Nitpick: 
```suggestion
    "reason": "Bug in fast any tap changing strategy in this edge case"
```

Also adding the reference to the issue here for future reference https://github.com/PowerGridModel/power-grid-model/issues/1215

File: tests/data/power_flow/automatic-tap-regulator/achievable-voltage-below-band-fast-any/params.json (hunk 2 above)
Comment (figueroa1395) [inline tests/data/power_flow/automatic-tap-regulator/achievable-voltage-below-band-fast-any/params.json]: Why use skip and not add handle the same as the other test case?

Also pointing at https://github.com/PowerGridModel/power-grid-model/issues/1215 for future reference.

**Comment by mgovers** [comment]
oh i see santiago also reviewed and some are duplicate as a result. will resolve the strictly duplicate ones and leave open the suggestion ones

File: tests/data/power_flow/automatic-tap-regulator/achievable-voltage-above-band-any/output.json
Comment (Jerry-Jinfeng-Guo) [reply tests/data/power_flow/automatic-tap-regulator/achievable-voltage-above-band-any/output.json:1]: Done. https://github.com/PowerGridModel/power-grid-model/pull/1214/commits/e2bb9b95a5fbd8fabe9ba3e4ab1fa73fae01858b Mistake.

File: tests/data/power_flow/automatic-tap-regulator/achievable-voltage-below-band-fast-any/params.json (hunk 2 above)
Comment (Jerry-Jinfeng-Guo) [reply tests/data/power_flow/automatic-tap-regulator/achievable-voltage-below-band-fast-any/params.json]: Removed. This one actually passes. https://github.com/PowerGridModel/power-grid-model/pull/1214/commits/e2bb9b95a5fbd8fabe9ba3e4ab1fa73fae01858b Mistake.

File: tests/data/power_flow/automatic-tap-regulator/achievable-voltage-above-band-fast-any/params.json (hunk 1 above)
Comment (Jerry-Jinfeng-Guo) [reply tests/data/power_flow/automatic-tap-regulator/achievable-voltage-above-band-fast-any/params.json]: Done. https://github.com/PowerGridModel/power-grid-model/pull/1214/commits/e2bb9b95a5fbd8fabe9ba3e4ab1fa73fae01858b .

File: tests/data/power_flow/automatic-tap-regulator/achievable-voltage-below-band-fast-any/params.json (hunk 2 above)
Comment (Jerry-Jinfeng-Guo) [reply tests/data/power_flow/automatic-tap-regulator/achievable-voltage-below-band-fast-any/params.json]: Removed. This one actually passes. https://github.com/PowerGridModel/power-grid-model/pull/1214/commits/e2bb9b95a5fbd8fabe9ba3e4ab1fa73fae01858b

**Review by figueroa1395 — APPROVED** [review]

File: tests/data/power_flow/automatic-tap-regulator/achievable-voltage-above-band-fast-any/params.json (hunk 1 above)
Comment (Jerry-Jinfeng-Guo) [reply tests/data/power_flow/automatic-tap-regulator/achievable-voltage-above-band-fast-any/params.json]: @mgovers Is this comment ok to resolve?
